components/component.go

```diff
@@ -86,6 +87,7 @@ type ComponentInterface interface {
 		owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, platform cluster.Platform, currentComponentStatus bool) error
 	Cleanup(ctx context.Context, cli client.Client, owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec) error
 	GetComponentName() string
+	GetComponentStatus() ([]status.ComponentReleaseStatus, error)
```

Comment:
It may be better to pass a `DataScienceCluster` instance to the method, so each component can set the related status according to the specif need.

Reply:
Thinking a little bit more, it may even be moved in the `ReconcileComponent` method (with some method signature changes)

Reply:
@lburgazzoli this approach didnt work as expected. I am unable to pass *dscv1.DataScienceCluster to ReconcileComponent method. getting error `import cycle not allowedgo list
Package v1 contains API Schema definitions for the datasciencecluster v1 API group`